ui/components/app/wallet-overview/coin-buttons.tsx

```diff
@@ -65,17 +74,21 @@ const CoinButtons = ({
   ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
   isBridgeChain,
   isBuyableChain,
+  isBuyableChainWithoutSigning = false,
   defaultSwapsToken,
   ///: END:ONLY_INCLUDE_IF
   classPrefix = 'coin',
 }: {
-  classPrefix?: string;
-  isBuyableChain: boolean;
-  isSigningEnabled: boolean;
+  chainId: `0x${string}` | CaipChainId | number;
   isSwapsChain: boolean;
+  isSigningEnabled: boolean;
+  ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
   isBridgeChain: boolean;
-  chainId: `0x${string}` | CaipChainId | number;
+  isBuyableChain: boolean;
+  isBuyableChainWithoutSigning?: boolean;
```

Comment:
I did introduce a new flag here, cause having `isSigningEnabled` set to `true` will enable the "Send" button, and for now we don't want that one to be enabled for BTC.

Instead of having some weird conditions that checks for `isEvm` and that kinds of things, I thought that having a dedicated flag would make things clearer.

Also, this will probably be removed when this PR is gonna land (will also required BTC support, but that would be new way of detecting buyable chain): https://github.com/MetaMask/metamask-extension/pull/24041

Reply:
Once #24041 is merged I will prepare a PR to update the logic for `isNativeTokenBuyable` to include a check if the user has selected a BTC account (see psuedo code below).

this will allow us to control the "buyability" of btc from our API and we can control UI elements based on that.

```
// will soon include logic for btc
shouldShowBuy ? (
  <RampsCard variant={RAMPS_CARD_VARIANT_TYPES.TOKEN} />
) : null
```

```
export const getIsNativeTokenBuyable = createSelector(
  [getCurrentChainId, getBuyableChains],
  (currentChainId, buyableChains) => {
    try {
      return buyableChains
        .filter(Boolean)
        .some(
          (network: AggregatorNetwork) => {
            // LOGIC NEEDED TO CHECK FOR BTC ACCOUNT SELECTION
            return String(network.chainId) === hexToDecimal(currentChainId),
           }
        );
    } catch (e) {
      return false;
    }
  },
);
```

Reply:
I'll add a comment to mention this new feature of dynamic buyable chains!